### PR TITLE
x.org: mirror mixin

### PR DESCRIPTION
**Pull request by ax3l**
According to my nightly CI/CD tests, x.org is another large provider of software in common build chains that is often down.

Added a hand-selected amount of mirrors that is well up-to-sync. Tested with `util-macros` that has a quite "recent" patch release.

Other packages to follow in an individual PR.

### CODE REVIEW

File: lib/spack/spack/build_systems/xorg.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @property
+    def urls(self):
+        self._ensure_xorg_mirror_path_is_set_or_raise()
+        return [
+            spack.util.url.join(m, self.xorg_mirror_path,
+                                resolve_href=True)
+            for m in self.base_mirrors
+        ]
+
+    def _ensure_xorg_mirror_path_is_set_or_raise(self):
```
Comment (tgamblin) [inline lib/spack/spack/build_systems/xorg.py:32]: Can you add a test that calls this function for some mock Xorg package and ensures that the right things are set?

Comment (ax3l) [reply lib/spack/spack/build_systems/xorg.py:32]: have it nearly ready, but am unsure how to create a mock `spack.package.PackageBase` object that @alalazo uses for those mirror mixin class interfaces.

**Comment by tgamblin** [comment]
This is awesome!

**Comment by ax3l** [comment]
I forward the thanks to @alalazo for the wonderful mirror support he added :)

**Pull request merged**